tests/python/pants_test/pantsd/pantsd_integration_test.py

```diff
@@ -434,158 +445,85 @@ def test_pantsd_multiple_parallel_runs(self):
             creator_handle.join().assert_success()
             waiter_handle.join().assert_success()
 
-    def _assert_pantsd_keyboardinterrupt_signal(self, signum, regexps=[], quit_timeout=None):
+    def _launch_waiter(self, workdir: str, config) -> Tuple[PantsJoinHandle, int, str]:
+        """Launch a process via pantsd that will wait forever for the a file to be created.
+
+        Returns the pid of the pantsd client, the pid of the waiting child process, and the file to
+        create to cause the waiting child to exit.
+        """
+        file_to_make = os.path.join(workdir, "some_magic_file")
+        waiter_pid_file = os.path.join(workdir, "pid_file")
```

Comment:
Seems to me like it would be a good idea to use the pathlib API consistently in new code, rather than using Pathlib some places and os.path.join others.

Reply:
I used to agree with this, but os.path seems to often be faster iirc, likely as it means always sticking with strings and avoiding type conversions.